kv/kvserver: TestGetChecksumNotSuccessfulExitConditions failed

kv/kvserver.TestGetChecksumNotSuccessfulExitConditions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9246019?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9246019?buildTab=artifacts#/) on release-22.1 @ [48becf7ef90056fc94a478e7d00f30d1c91af07a](https://github.com/cockroachdb/cockroach/commits/48becf7ef90056fc94a478e7d00f30d1c91af07a):


```
=== RUN   TestGetChecksumNotSuccessfulExitConditions
    replica_consistency_test.go:117: checksum computation did not start in time for (ID = f145b066-ecc6-4f33-9d06-5fe329c9f61b)
--- FAIL: TestGetChecksumNotSuccessfulExitConditions (1.16s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGetChecksumNotSuccessfulExitConditions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25938

--- a/pkg/kv/kvserver/replica_consistency_test.go
+++ b/pkg/kv/kvserver/replica_consistency_test.go
@@ -108,8 +108,7 @@ func TestGetChecksumNotSuccessfulExitConditions(t *testing.T) {
 	require.Nil(t, rc.Checksum)
 
 	// Need to reset the context, since we deadlined it above.
-	ctx, cancel = context.WithTimeout(context.Background(), 100*time.Millisecond)
-	defer cancel()
+	ctx = context.Background()
 	// Next condition, node should quiesce.
 	tc.repl.store.Stopper().Quiesce(ctx)
 	rc, err = tc.repl.getChecksum(ctx, uuid.FastMakeV4())

